eng/targets/GeneratePkgDef.targets

```diff
@@ -0,0 +1,221 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project>
+  <!--
+    This file must be imported after targets from NuGet packages have been imported (e.g. from Directory.Build.targets).
+    It overwrites GeneratePkgDef defined by VSSDK NuGet package targets.
+    
+    GeneratePkgDef target defined below generates .pkgdef file based on PkgDefEntry items.
+    Unlike the VSSDK implementation it does not load the project assembly and does not use RegistrationAttributes.
+    The following kinds of PkgDefEntry are recognized (based on Kind metadata of the item):
+    
+      1) PackageRegistration 
+         Corresponds to Microsoft.VisualStudio.Shell.PackageRegistrationAttribute.
+         Project specifies this entry explicitly.
+         
+      2) InstalledProduct
+         Emits $RootKey$\InstalledProducts entry.
+         Project specifies this entry explicitly.
+         
+      3) BindingRedirect
+         Emits $RootKey$\RuntimeConfiguration\dependentAssembly\bindingRedirection entry.
+         
+         Project may specify <PkgDefEntry>BindingRedirect</PkgDefEntry> on any item that contributes to ReferencePath item group.
+         (e.g. ProjectReference), or on NuGetPackageToIncludeInVsix items. These items will be automatically included in PkgDefEntry items.
+         
+      4) CodeBase
+         Emits $RootKey$\RuntimeConfiguration\dependentAssembly\codeBase entry.
+         
+         Project may specify <PkgDefEntry>CodeBase</PkgDefEntry> on any item that contributes to ReferencePath item group.
+         (e.g. ProjectReference), or on NuGetPackageToIncludeInVsix items. These items will be automatically included in PkgDefEntry items.
+         
+      5) FileContent
+         Merges the content of the file whose path is specified in the identity of the item to the generated pkgdef file.
+         The path may be relative to the project being built.
```

Comment:
Yes, the VSSDK expects a single pkgdef and flows it thru various places. I only added the merging because the other way was not working consistently. 
VSSDK allows you to add pkgdef file to the VSIX but it's just an arbitrary file at that point that doesn't flow trhu build groups etc.
At runtime the extension manager indeed applies all the pkgdef files contained in the VSIX